Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -556,13 +444,32 @@ void ArgsParser::add_option(Vector<size_t>& values, char const* help_string, cha
             });
 
             return parsed_all_values;
-        },
+        });
+}
+
+template void ArgsParser::add_option(char const*&, char const*, char const*, char, char const*, OptionHideMode);
+template void ArgsParser::add_option(StringView&, char const*, char const*, char, char const*, OptionHideMode);
+template void ArgsParser::add_option(String&, char const*, char const*, char, char const*, OptionHideMode);
+
+template<bool RequireArgument>
+void ArgsParser::add_option_helper(char const* help_string, char const* long_name, char short_name, char const* value_name, ArgsParser::OptionHideMode hide_mode, Function<bool(char const*)> accept_value)
```

Comment:
Yeah no, please just make the template argument a normal one, it's really not worth generating two instances and then merging them just to pass a single bool in a very cold function.

Reply:
I did it because to indicate to the reader that they are compile-time argument, but wasn't really sure of the design and really wanted a review haha.